filter/filter.go

```diff
@@ -112,20 +112,43 @@ func CompileFieldCompare(node ast.CompareField, val zeek.Value) (Filter, error)
 	return combine(resolver, comparison), nil
 }
 
-func EvalAny(eval zeek.Predicate) Filter {
-	return func(p *zson.Record) bool {
-		it := p.ZvalIter()
-		for _, c := range p.Type.Columns {
+func EvalAny(eval zeek.Predicate, recursive bool) Filter {
+	if !recursive {
+		return func(r *zson.Record) bool {
+			it := r.ZvalIter()
+			for _, c := range r.Type.Columns {
+				val, _, err := it.Next()
+				if err != nil {
+					return false
+				}
+				if eval(zeek.TypedEncoding{c.Type, val}) {
+					return true
+				}
+			}
+			return false
+		}
+	}
+
+	var fn func(v zval.Encoding, cols []zeek.Column) bool
+	fn = func(v zval.Encoding, cols []zeek.Column) bool {
```

Comment:
Just say `fn := ...`.

Reply:
That's how I wrote it at first, but then fn wasn't visible a few lines below...